File: Lv0/제곱수판별하기/Solution.js
function solution(n) {
    let answer = 0;
    let sqrt = Number.isInteger(Math.sqrt(n))
    if(sqrt){
        answer = 1;
    }else{
        answer = 2;
    }
    return answer;
}

// Math.sqrt(n) -> n의 제곱근을 구한다
//Number.isInteger() -> 결과가 정수인지 아닌지 확인한다.

File: Lv0/제곱수판별하기/Solution.java
package Lv0.제곱수판별하기;

class Solution {
    public int solution(int n) {
        int answer = 0;
        // 제곱근을 구한다.
        double sqrt = Math.sqrt(n);
        if(sqrt == (int)sqrt){
            answer = 1;
        }else{
            answer = 2;
        }
        return answer;
    }
}